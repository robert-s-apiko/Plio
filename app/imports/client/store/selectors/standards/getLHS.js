import { createSelector } from 'reselect';

import getSortedStandardsByFilter from './getSortedStandardsByFilter';
import getFilteredStandards from './getFilteredStandards';
import { getIsModalOpened } from '../dataImport';
import { getOrganizationId } from '../organizations';
import {
  getSearchText,
  getFilter,
  getAnimating,
  getUrlItemId,
  getUserId,
} from '../global';
import getSearchMatchText from '../../../../api/helpers/getSearchMatchText';

/*
selector(
  standards: Array,
  filteredStandards: Array,
  isModalOpened: Boolean,
  organizationId: String,
  searchText: String,
  filter: Number,
  animating: Boolean,
  urlItemId: String,
  userId: String
) => Object
*/

const selector = (
  standards,
  filteredStandards,
  isModalOpened,
  organizationId,
  searchText,
  filter,
  animating,
  urlItemId,
  userId,
) => ({
  standards,
  filteredStandards,
  organizationId,
  searchText,
  filter,
  animating,
  urlItemId,
  userId,
  isDataImportModalOpened: isModalOpened,
  searchResultsText: getSearchMatchText(searchText, standards.length),
});

export default createSelector([
  getSortedStandardsByFilter,
  getFilteredStandards,
  getIsModalOpened,
  getOrganizationId,
  getSearchText,
  getFilter,
  getAnimating,
  getUrlItemId,
  getUserId,
], selector);
