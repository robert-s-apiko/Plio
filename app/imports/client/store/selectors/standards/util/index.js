export { default as withUncategorized } from './withUncategorized';
