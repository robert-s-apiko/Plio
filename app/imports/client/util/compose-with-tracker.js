import { Tracker } from 'meteor/tracker';
import { compose } from 'ramda';

function getTrackerLoader(loaderFunc) {
  return (props, onData, env) => {
    let trackerCleanup = () => null;

    const handler = Tracker.nonreactive(() =>
      Tracker.autorun(() => {
        // Store clean-up function if provided.
        trackerCleanup = loaderFunc(props, onData, env) || (() => null);
      }));

    return () => {
      trackerCleanup();
      return handler.stop();
    };
  };
}

function composeWithTracker(loadFunc) {
  return function (component) {
    return compose(getTrackerLoader(loadFunc))(component);
  };
}

export default composeWithTracker;
