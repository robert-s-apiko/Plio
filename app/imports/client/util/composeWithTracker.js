import { Tracker } from 'meteor/tracker';
import { compose } from '@storybook/react-komposer';

function getTrackerLoader(loaderFunc) {
  return (props, onData, env) => {
    let trackerCleanup = () => null;

    const handler = Tracker.nonreactive(() =>
      Tracker.autorun(() => {
        // Store clean-up function if provided.
        trackerCleanup = loaderFunc(props, onData, env) || (() => null);
      }));

    return () => {
      trackerCleanup();
      return handler.stop();
    };
  };
}

function composeWithTracker(loadFunc, options) {
  return function (component) {
    // explicitly pass { withRef: false }
    // otherwise react will throw error
    // that the component tries to access
    // a ref of the stateless component
    const opts = Object.assign({ withRef: false }, options);

    return compose(getTrackerLoader(loadFunc), opts)(component);
  };
}

export default composeWithTracker;
