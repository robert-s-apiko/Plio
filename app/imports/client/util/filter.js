import { ifElse, equals, compose, filter, nthArg, complement } from 'ramda';

import { isDeleted } from './object-relation';

// (filterDeleted: Boolean, items: Array) => Array
export const filterByDeleted = ifElse(
  equals(true),
  compose(filter(isDeleted), nthArg(1)),
  compose(filter(complement(isDeleted)), nthArg(1)),
);
