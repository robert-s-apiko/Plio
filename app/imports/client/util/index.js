export { default as composeWithTracker } from './composeWithTracker';
