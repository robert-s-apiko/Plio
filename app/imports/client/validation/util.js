import { FORM_ERROR } from 'final-form';
import { forEachObjIndexed, curry, isNil } from 'ramda';

export const required = curry((label, value) => !isNil(value) ? undefined : `${label} required`);

export const validate = rules => (values) => {
  const errors = new Set();

  forEachObjIndexed((value, key) => {
    if (rules[key]) {
      const error = rules[key](value, values);
      if (error) errors.add(error);
    }
  }, values);

  if (!errors.size) return undefined;

  return { [FORM_ERROR]: [...errors].join('\n') };
};
