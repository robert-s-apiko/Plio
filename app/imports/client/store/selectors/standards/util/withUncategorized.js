import { set, filter, propEq, append, compose, map } from 'ramda';
import { lenses, getStandardsLength } from '../../../../util';

// (prop: String, standards: Array) => (item: Object) => Object
export const setFilteredStandards = (prop, standards) => item => set(
  lenses.standards,
  filter(propEq(prop, item._id), standards),
  item,
);

// (prop: String, uncategorized: Object, standards: Array, items: Array) => Array
export default (prop, uncategorized, standards, items) => compose(
  filter(getStandardsLength),
  append(uncategorized),
  map(setFilteredStandards(prop, standards)),
)(items);
