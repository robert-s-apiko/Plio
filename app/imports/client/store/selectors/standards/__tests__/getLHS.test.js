import getLHS from '../getLHS';
import { getInitialState } from '../../../util/tests';

describe('getLHS', () => {
  it('matches snapshot', () => {
    const state = getInitialState({});

    expect(getLHS(state)).toMatchSnapshot();
  });
});
