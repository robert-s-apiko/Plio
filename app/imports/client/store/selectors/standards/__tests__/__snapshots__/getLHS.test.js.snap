// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getLHS matches snapshot 1`] = `
Object {
  "animating": undefined,
  "filter": 1,
  "filteredStandards": Array [
    Object {
      "_id": 1,
    },
    Object {
      "_id": 2,
    },
    Object {
      "_id": 5,
    },
  ],
  "isDataImportModalOpened": false,
  "organizationId": 7,
  "searchResultsText": "",
  "searchText": "",
  "standards": Array [
    Object {
      "_id": 1,
    },
    Object {
      "_id": 2,
    },
    Object {
      "_id": 5,
    },
  ],
  "urlItemId": 1,
  "userId": 1,
}
`;
